### PR TITLE
Added cypress test for new patient creation

### LINKED ISSUE #4233: Cypress: New Patient Creation

### PR DESCRIPTION
**Pull request by ShivamJhaa**
### <samp>🤖 Generated by Copilot at 06c480f</samp>

Updated patient creation and dashboard tests to match new UI and added logging for phone number fields. The `cypress/e2e/patient_spec/patient_crud.cy.ts` file was modified to use new UI elements and randomize test data. The `src/Components/Patient/PatientRegister.tsx` file was modified to log phone number inputs and validations.

## Proposed Changes

- Fixes #4233 
- Change 1
- Change 2
- More?

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### <samp>🤖 Generated by Copilot at 06c480f</samp>

*  Update patient creation flow to match new UI design ([link](https://github.com/coronasafe/care_fe/pull/5549/files?diff=unified&w=0#diff-85edc6c535fd4c7cef250d5a68bb562c3fd75a82654e2de0b0227413d5840bfaL26-R79), [link](https://github.com/coronasafe/care_fe/pull/5549/files?diff=unified&w=0#diff-85edc6c535fd4c7cef250d5a68bb562c3fd75a82654e2de0b0227413d5840bfaL55-R92), [link](https://github.com/coronasafe/care_fe/pull/5549/files?diff=unified&w=0#diff-09e14a8b77195fb6628b716eb09a71a996a55f559ddc5c3ef19d3c54948a4775R612-R613), [link](https://github.com/coronasafe/care_fe/pull/5549/files?diff=unified&w=0#diff-09e14a8b77195fb6628b716eb09a71a996a55f559ddc5c3ef19d3c54948a4775R624-R625))
* Assign random values to phone number and emergency phone number variables using helper function `makePhoneNumber` ([link](https://github.com/coronasafe/care_fe/pull/5549/files?diff=unified&w=0#diff-85edc6c535fd4c7cef250d5a68bb562c3fd75a82654e2de0b0227413d5840bfaL5-R10))
* Change URL assertions to match the updated navigation paths ([link](https://github.com/coronasafe/care_fe/pull/5549/files?diff=unified&w=0#diff-85edc6c535fd4c7cef250d5a68bb562c3fd75a82654e2de0b0227413d5840bfaL55-R92), [link](https://github.com/coronasafe/care_fe/pull/5549/files?diff=unified&w=0#diff-85edc6c535fd4c7cef250d5a68bb562c3fd75a82654e2de0b0227413d5840bfaL71-R103), [link](https://github.com/coronasafe/care_fe/pull/5549/files?diff=unified&w=0#diff-85edc6c535fd4c7cef250d5a68bb562c3fd75a82654e2de0b0227413d5840bfaL89-R120))
* Log patient URL, phone number and emergency phone number for debugging purposes ([link](https://github.com/coronasafe/care_fe/pull/5549/files?diff=unified&w=0#diff-85edc6c535fd4c7cef250d5a68bb562c3fd75a82654e2de0b0227413d5840bfaL71-R103), [link](https://github.com/coronasafe/care_fe/pull/5549/files?diff=unified&w=0#diff-09e14a8b77195fb6628b716eb09a71a996a55f559ddc5c3ef19d3c54948a4775R612-R613), [link](https://github.com/coronasafe/care_fe/pull/5549/files?diff=unified&w=0#diff-09e14a8b77195fb6628b716eb09a71a996a55f559ddc5c3ef19d3c54948a4775R624-R625))

### CODE REVIEW

File: cypress/e2e/patient_spec/patient_crud.cy.ts [hunk 1]
```diff
@@ -2,9 +2,12 @@ import { cy, describe, before, beforeEach, it, afterEach } from "local-cypress";
 
 const username = "devdistrictadmin";
 const password = "Coronasafe@123";
-const phone_number = "9" + parseInt((Math.random() * 10 ** 9).toString());
-const emergency_phone_number = "9430123487";
-const yearOfBirth = "1999";
+const makePhoneNumber = () =>
+  "99" + Math.floor(Math.random() * 99999999).toString();
```
Comment (rithviknishad) [inline cypress/e2e/patient_spec/patient_crud.cy.ts]: Wouldn't this cause random fails?

Say the math.random generated 0.0000001 or some low values, wouldn't this result in a decrease number of digits?

Comment (ShivamJhaa) [reply cypress/e2e/patient_spec/patient_crud.cy.ts]: True, reverted the change.
Thanks!

**Comment by ShivamJhaa** [comment]
@rithviknishad @nihal467 Can you please review the PR?
Thanks!

**Comment by ShivamJhaa** [comment]
@khavinshankar Its strange that it passes successfully locally but fails on CI:
Screenshot from 2023-06-01 16-08-02

Can you please re-run the test?

**Comment by ShivamJhaa** [comment]
@khavinshankar Now the test is passing on Cl as well.

**Comment by nihal467** [comment]
@khavinshankar can you review this ?

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**